lib/modules/manager/maven/extract.ts

```diff
@@ -542,6 +543,26 @@ export async function extractAllPackageFiles(
 ): Promise<PackageFile[]> {
   const packages: PackageFile[] = [];
   const additionalRegistryUrls: string[] = [];
+  const homedir = os.homedir();
+  const homeSettingsPath = upath.join(homedir, '.m2', 'settings.xml');
+  try {
+    const homeSettingsContent = await readSystemFile(homeSettingsPath, 'utf8');
+    if (homeSettingsContent) {
+      const homeRegistries = extractRegistries(homeSettingsContent);
+      if (homeRegistries) {
+        logger.debug(
+          { homeRegistries, homeSettingsPath },
+          'Found registryUrls in $HOME/.m2/settings.xml',
+        );
+        additionalRegistryUrls.push(...homeRegistries);
+      }
+    }
+  } catch (err) {
+    logger.debug(
+      { homeSettingsPath, err },
+      'No settings.xml found in $HOME/.m2 or error reading file',
```

Comment:
Won't this log an error (stack trace!) for the majority of users who aren't using a home m2 on the renovate machine?